state/src/item/metadata.rs

```diff
@@ -93,6 +95,14 @@ impl Metadata {
         self.params = Some(params);
     }
 
+    pub fn term_params(&self) -> Option<&CommonParams> {
+        self.term_params.as_ref()
+    }
+
+    pub fn snapshot_term_params(&mut self) {
```

Comment:
I think there is a more proper name than `snapshot_term_params`.

Reply:
I know `snapshot` is a little bit confusing. It collides with the name of the snapshot sync feature and it is awkward to use it as a verb to me, but I can't think of a better one. Can you suggest a good one? How about `freeze_`?

Reply:
Would you mind if we change the name later? @Remagpie is dependant on this PR.

Reply:
Changing the name is not a big deal in my work. You can change it now if you want.